src/radiation/moments.cpp

```diff
@@ -804,29 +1124,94 @@ TaskStatus MomentFluidSource(T *rc, Real dt, bool update_fluid) {
 
           // Add source corrections to conserved fluid variables
           if (update_fluid) {
-            v(iblock, cye, k, j, i) -= sdetgam * 0.0;
+            if (cye > 0) {
+              v(iblock, cye, k, j, i) -= sdetgam * 0.0;
+            }
+#if USE_VALENCIA
             v(iblock, ceng, k, j, i) -= sdetgam * dE;
+#else
+            // TODO(BRR) This is not correct
+            v(iblock, ceng, k, j, i) += alpha * sdetgam * dE;
+#endif
             v(iblock, cmom_lo + 0, k, j, i) -= sdetgam * cov_dF(0);
             v(iblock, cmom_lo + 1, k, j, i) -= sdetgam * cov_dF(1);
             v(iblock, cmom_lo + 2, k, j, i) -= sdetgam * cov_dF(2);
           }
-        }
+        } // for ispec
+
+        /* for (int ispec = 0; ispec < num_species; ++ispec) {
```

Comment:
I'll go back and make the source update algorithm a runtime option, probably a templated `FluidSourceImpl` function like we do elsewhere.